code/modules/reagents/chemical_research/Chemical-Research.dm

```diff
@@ -24,21 +24,59 @@ var/global/datum/chemical_data/chemical_data = new /datum/chemical_data/
 /datum/chemical_data/proc/save_document(var/obj/item/paper/research_report/R, var/document_type, var/title)
 	if(!research_documents["[document_type]"])
 		research_documents["[document_type]"] = list()
-	var/list/new_document[0]
-	new_document["[title]"] = R
-	research_documents["[document_type]"] += new_document
+	var/save_time = worldtime2text()
+
+	var/list/new_document = list(
+		"document_title"=title,
+		"time"=save_time,
+		"document"=R
+	)
+	research_documents["[document_type]"] += list(new_document)
+
+/datum/chemical_data/proc/get_report(var/doc_type, var/doc_title)
+	var/obj/item/paper/research_report/report = null
+	for(var/i in chemical_data.research_documents[doc_type])
+		if(i["document_title"] == doc_title)
+			report = i["document"]
+			break
```

Comment:
use a detailed variable name for i here.

Reply:
Resolved this.